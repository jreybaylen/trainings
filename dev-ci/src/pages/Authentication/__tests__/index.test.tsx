import React from 'react'
import userEvent from '@testing-library/user-event'
import { render, screen } from '@testing-library/react'

import AuthenticationPage from '../'

describe('Authentication Page', () => {
    const handleSubmit = jest.fn()

    beforeEach(() => {
        render(<AuthenticationPage onSubmit={ handleSubmit } />)
    })

    it('should render without crashing', () => {
        expect(screen.getByText('Hello Authentication!')).toMatchSnapshot()
    })

    it('should call "onClick" prop once the user tapped the button', async () => {
        await userEvent.click(screen.getByText('Click Me!'))
        expect(handleSubmit).toHaveBeenCalled()
    })
})