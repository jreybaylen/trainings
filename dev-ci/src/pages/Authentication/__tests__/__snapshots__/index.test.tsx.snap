// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication Page should render without crashing 1`] = `
<div>
  Hello Authentication!
</div>
`;
